sdk/iot/Azure.Iot.Hub.Service/samples/IotHubClientSamples/MehodInvocationSamples.cs

```diff
@@ -0,0 +1,236 @@
+        public static readonly Random Random = new Random();
+
+        public MethodInvocationSamples(IotHubServiceClient client)
+        {
+            IoTHubServiceClient = client;
+        }
+
+        public async Task RunSampleAsync()
+        {
+            string deviceId = $"device{Random.Next(MaxRandomValue)}";
+            string moduleId = $"module{Random.Next(MaxRandomValue)}";
+
+            // Create a DeviceIdentity.
+            await CreateDeviceIdentityAsync(deviceId);
+
+            // Create a ModuleIdentity.
+            await CreateModuleIdentityAsync(deviceId, moduleId);
+
+            // Invoke method on device
+            await InvokeMethodOnDeviceAsync(deviceId);
+
+            // Invoke method on module
+            await InvokeMethodOnModuleAsync(deviceId, moduleId);
+
+            // Delete the device.
+            await DeleteDeviceIdentityAsync(deviceId);
+        }
+
+        /// <summary>
+        /// Invokes a method on the module.
+        /// </summary>
+        /// <param name="deviceId">Unique identifier of the device to be updated.</param>
+        /// <param name="moduleId">Unique identifier of the module to be updated.</param>
+        public async Task InvokeMethodOnModuleAsync(string deviceId, string moduleId)
+        {
+            SampleLogger.PrintHeader("INVOKE REBOOT METHOD ON A MODULE");
+
+            try
+            {
+                #region Snippet:IotInvokeMethodOnModule
+
+                var request = new CloudToDeviceMethodRequest
+                {
+                    MethodName = "reboot",
+                };
+                Response<CloudToDeviceMethodResponse> response = await IoTHubServiceClient.Modules.InvokeMethodAsync(deviceId, moduleId, request);
+
+                SampleLogger.PrintSuccess($"\t- Method invoked");
+
+                #endregion Snippet:IotInvokeMethodOnModule
+            }
+            catch (Exception ex)
+            {
+                // Try to cleanup before exiting with fatal error.
+                await CleanupHelper.DeleteAllDevicesInHubAsync(IoTHubServiceClient);
+                SampleLogger.FatalError($"Failed to invoke method on module due to:\n{ex}");
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Invokes a method on the device.
+        /// </summary>
+        /// <param name="deviceId">Unique identifier of the device to be updated.</param>
+        public async Task InvokeMethodOnDeviceAsync(string deviceId)
+        {
+            SampleLogger.PrintHeader("INVOKE REBOOT METHOD ON A DEVICE");
+
+            try
+            {
+                #region Snippet:IotInvokeMethodOnDevice
+
+                var request = new CloudToDeviceMethodRequest
+                {
+                    MethodName = "reboot",
+                };
+                Response<CloudToDeviceMethodResponse> response = await IoTHubServiceClient.Devices.InvokeMethodAsync(deviceId, request);
```

Comment:
I'd add a bit more to the handling of the return value here. A typical customer should look at the status code in the CloudToDeviceMethodResponse type, and they may have a response payload in that same object. Just printing out those two values (or 1 value if payload is null) would add a lot to this sample